torch/types.py

```diff
@@ -34,7 +34,7 @@
 # Meta-type for "device-like" things.  Not to be confused with 'device' (a
 # literal device object).  This nomenclature is consistent with PythonArgParser.
 # None means use the default device (typically CPU)
-Device = Union[_device, str, _int, None]
+Device = Optional[Union[_device, str, _int]]
```

Comment:
Optional[DeviceTypeLike] ? Or that would be a circular import?

Reply:
I didn't actually try it but yeah being a bit cautious I guess since this separate definition already existed. Just adding `Optional` in the import required adding to `test/allowlist_for_publicAPI.json`.

Reply:
Nothing should be added to that list, either add a proper `__all__` to this file or `del` this one at the end.

Reply:
del'd the Optional and removed from the allowlist. I also tested with Optional[DeviceLikeType] here in types.py and confirmed it does lead to circular import errors.